examples/url-shortener/src/lib.rs

```diff
@@ -0,0 +1,64 @@
+#[macro_use]
+extern crate rocket;
+
+use rocket::{
+    response::{status::BadRequest, Redirect},
+    routes, Build, Rocket, State,
+};
+use serde::Serialize;
+use shuttle_service::Factory;
+use sqlx::{Executor, FromRow, PgPool};
+
+#[macro_use]
+extern crate shuttle_service;
+
+struct AppState {
+    pool: PgPool,
+}
+
+#[derive(Serialize, FromRow)]
+struct Url {
+    pub id: String,
+    pub url: String,
+}
+
+#[get("/<id>")]
+async fn redirect(id: String, state: &State<AppState>) -> Result<Redirect, BadRequest<String>> {
+    let url: Url = sqlx::query_as("SELECT * FROM urls WHERE id = $1")
+        .bind(id)
+        .fetch_one(&state.pool)
+        .await
+        .map_err(|e| BadRequest(Some(e.to_string())))?;
+
+    Ok(Redirect::to(url.url))
+}
+
+#[post("/", data = "<url>")]
+async fn shorten(url: String, state: &State<AppState>) -> Result<String, BadRequest<String>> {
+    let id = &nanoid::nanoid!(6);
+
+    sqlx::query("INSERT INTO urls(id, url) VALUES ($1, $2)")
+        .bind(id)
+        .bind(&url)
```

Comment:
This just takes whatever the user gives and stores it, regardless of if it's valid or not. Should we use something like the `url` crate to validate? That way we can also ensure we force user input to be an absolute url.